### DISCUSSION TITLE
Getting error : NoMethodError (undefined method `passphrase_password_path'

### DISCUSSION THREAD

**Discussion by amelerowicz-oceanex-com** [pglombardo/PasswordPusher#1447] [Q&A] [answered]
Good day, 

I am seeking for help regarding a very specific issue we are experiencing. 
We have been using pwpush for a while now without any issue and we are really happy with it. However, we wanted to get the passphrase lockdown feature but unfortunately since we deployed the latest version, we are experiencing Errors 500 while using this function.

Here is the docker command we are using to deploy the docker container: 

`docker run --detach --name pwpush --restart=always --env DATABASE_URL=mysql2://dbu_pwpush:xxxxxxxxxx@mariadb:3306/db_pwpush --env PWPUSH_MASTER_KEY=xxxxxxxxxx --env PWP__THEME=zephyr --env PWP_PRECOMPILE=true -p "5100:5100" --network="backend" -v pwpush_config:/opt/PasswordPusher/config -v pwpush_logos:/opt/PasswordPusher/public/logos -v pwpush_storage:/opt/PasswordPusher/storage:rw pglombardo/pwpush-mysql:latest`

As you can see, there is a backend network and we are using a nginx reverse proxy container to get access to the web interface but even on the bridge network on port 5100, it seems that I get the same's errors while retrieving the password, URL or file:

_More information at https://github.com/enriclluelles/route_translator/pull/285
 (called from block (2 levels) in preview at /opt/PasswordPusher/app/controllers/passwords_controller.rb:253)
DEPRECATION WARNING: `i18n_use_slash_separator` set to \`false\` is deprecated and will be
removed in the next major release of Route Translator to match
Rails' ActiveRecord nested model syntax.

More information at https://github.com/enriclluelles/route_translator/pull/285
 (called from custom_set_locale_from_url at /opt/PasswordPusher/app/controllers/application_controller.rb:8)
DEPRECATION WARNING: `i18n_use_slash_separator` set to `false` is deprecated and will be
removed in the next major release of Route Translator to match
Rails' ActiveRecord nested model syntax.

More information at https://github.com/enriclluelles/route_translator/pull/285
 (called from block in raw_secret_url at /opt/PasswordPusher/app/helpers/application_helper.rb:26)
DEPRECATION WARNING: `i18n_use_slash_separator` set to `false` is deprecated and will be
removed in the next major release of Route Translator to match
Rails' ActiveRecord nested model syntax.

More information at https://github.com/enriclluelles/route_translator/pull/285
 (called from custom_set_locale_from_url at /opt/PasswordPusher/app/controllers/application_controller.rb:8)
F, [2023-09-19T21:54:21.819464 #208] FATAL -- : [85533f2d-1aae-4df9-ae76-9f48aceef751]   
[85533f2d-1aae-4df9-ae76-9f48aceef751] NoMethodError (undefined method `passphrase_password_path' for #<PasswordsController:0x00000000071340>):
[85533f2d-1aae-4df9-ae76-9f48aceef751]   
[85533f2d-1aae-4df9-ae76-9f48aceef751] app/controllers/passwords_controller.rb:72:in `block (2 levels) in show'
[85533f2d-1aae-4df9-ae76-9f48aceef751] app/controllers/passwords_controller.rb:71:in `show'
[85533f2d-1aae-4df9-ae76-9f48aceef751] app/controllers/application_controller.rb:14:in `custom_set_locale_from_url'
DEPRECATION WARNING: `i18n_use_slash_separator` set to `false` is deprecated and will be
removed in the next major release of Route Translator to match
Rails' ActiveRecord nested model syntax.

More information at https://github.com/enriclluelles/route_translator/pull/285
 (called from custom_set_locale_from_url at /opt/PasswordPusher/app/controllers/application_controller.rb:8)
F, [2023-09-19T21:54:53.179627 #208] FATAL -- : [feb6c267-0e01-4e58-aaed-b9b8c79e3fb3]   
[feb6c267-0e01-4e58-aaed-b9b8c79e3fb3] NoMethodError (undefined method `passphrase_password_path' for #<PasswordsController:0x00000000071408>):
[feb6c267-0e01-4e58-aaed-b9b8c79e3fb3]   
[feb6c267-0e01-4e58-aaed-b9b8c79e3fb3] app/controllers/passwords_controller.rb:72:in `block (2 levels) in show'
[feb6c267-0e01-4e58-aaed-b9b8c79e3fb3] app/controllers/passwords_controller.rb:71:in `show'
[feb6c267-0e01-4e58-aaed-b9b8c79e3fb3] app/controllers/application_controller.rb:14:in `custom_set_locale_from_url'_

I am not sure if this a bug or a misconfiguration but I have recreated the database from scratch, but also the volumes, the settings.yml,  the finally the container, without success. 

Is anyone able to provide me guidance on where to look at ?

Thank you,

Alexandre

**Comment by amelerowicz-oceanex-com** [comment, marked as answer]
I figured it out.... redirecting the settings folder is a bad idea... whenever you update the app, the files in this folder are not updated (by design)

**Reply by pglombardo** [reply]
That's great to hear that it's resolved now @amelerowicz-oceanex-com.
